### PR TITLE
docs: memory requirements

### PR DESCRIPTION
**Pull request by dandv**
Tried Gatsby on a small project, on a VPS with 512MB RAM, and it crashed with a memory allocation error.

Also, fixed the link to the style guide.

### CODE REVIEW

File: docs/docs/basic-hardware-software-requirements.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
...
 ---
 
+Gatsby requires at least 1GB of RAM.
+
+------
+
 This is a stub. Help our community expand it.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your pull request gets accepted.
+Please use the [Gatsby Style Guide](/docs/contributing/gatsby-style-guide.md) to ensure your pull request gets accepted.
```
Comment (LekoArts) [inline docs/docs/basic-hardware-software-requirements.md:11]: The docs are supposed to be read on gatsbyjs.org so please revert this change :)

**Comment by KyleAMathews** [comment]
Curious how big was your Gatsby site? I've seen memory usage in the past during builds on small sites be < 250mb but mem usage does grow as the size of your site grows.

**Comment by dandv** [comment]
@KyleAMathews: about 30 pages: index, about, faq, terms, and ~25 generated from GraphQL.

**Comment by AishaBlake** [comment]
Hey, @dandv! Thanks so much for the PR. Since the memory requirement is variable depending on the site, I don't think it makes sense to add a hard and fast rule to the docs like this. If you notice this again, please feel free to file an issue.
